irmago_test.go

```diff
@@ -798,3 +799,228 @@ func TestPrivateKeyRings(t *testing.T) {
+		ID: "testwizard",
+		Contents: IssueWizardContents{
+			{{
+				credwizarditem("scheme.issuer.c"),
+			}},
+			{{{
+				Type:       IssueWizardItemTypeCredential,
+				Credential: credidptr("scheme.issuer.e"),
+				Text:       &TranslatedString{"en": "custom description of credential e"},
+			}}},
+			{{{
+				Type: IssueWizardItemTypeWebsite,
+				URL:  &TranslatedString{"en": "https://example.com"},
+			}}},
+		},
+	}
+
+	contents, err := wizard.Choose(conf, nil)
+	require.NoError(t, err)
+	require.Equal(t,
+		[]IssueWizardItem{
+			credwizarditem("scheme.issuer.a"),
+			credwizarditem("scheme.issuer.b"),
+			credwizarditem("scheme.issuer.d"),
+			credwizarditem("scheme.issuer.c"),
+			wizard.Contents[1][0][0],
+			wizard.Contents[2][0][0],
+		},
+		contents,
+	)
+}
+
+func TestWizardFromScheme(t *testing.T) {
+	conf := parseConfiguration(t)
+	wizard := conf.Requestors["localhost"].Wizards["testwizard"]
+
+	var expected []IssueWizardItem
+	require.NoError(t, json.Unmarshal(
+		[]byte(`[{"type":"credential","credential":"irma-demo.MijnOverheid.fullName","header":{"en":"Full name","nl":"Volledige naam"},"text":{"en":"Lorem ipsum dolor sit amet, consectetur adipiscing elit.","nl":"Lorem ipsum dolor sit amet, consectetur adipiscing elit."},"label":{"en":"Get name data","nl":"Haal naamgegevens op"}},{"type":"credential","credential":"irma-demo.MijnOverheid.singleton"},{"type":"session","credential":"irma-demo.RU.studentCard","header":{"en":"Student Card","nl":"Studentpas"},"text":{"en":"Lorem ipsum dolor sit amet, consectetur adipiscing elit.","nl":"Lorem ipsum dolor sit amet, consectetur adipiscing elit."},"label":{"en":"Get Student Card","nl":"Haal studentpas op"},"sessionUrl":"https://example.com/getsession"},{"type":"website","credential":"irma-demo.stemmen.stempas","header":{"en":"Voting Card","nl":"Stempas"},"text":{"en":"Lorem ipsum dolor sit amet, consectetur adipiscing elit.","nl":"Lorem ipsum dolor sit amet, consectetur adipiscing elit."},"label":{"en":"Get Voting Card","nl":"Haal stempas op"},"url":{"en":"https://example.com/en","nl":"https://example.com/nl"},"inapp":true}]`),
+		&expected,
+	))
+
+	contents, err := wizard.Choose(conf, nil)
+	require.NoError(t, err)
+	require.Equal(t, expected, contents)
+
+	True := true
+	wizard.ExpandDependencies = &True
+	contents, err = wizard.Choose(conf, nil)
+	require.NoError(t, err)
+	require.Equal(t,
+		append([]IssueWizardItem{credwizarditem("irma-demo.MijnOverheid.root")}, expected...),
+		contents,
+	)
+}
+
+func TestCredentialDependencies_UnmarshalXML(t *testing.T) {
+	xmlbts := []byte(`<Dependencies>
+		<And>
+			<Or>
+				<CredentialType>a</CredentialType>
+				<CredentialType>b</CredentialType>
+			</Or>
+			<Or>
+				<CredentialType>c</CredentialType>
+			</Or>
+		</And>
+		<And>
+			<Or>
+				<CredentialType>d</CredentialType>
+			</Or>
+			<Or>
+				<CredentialType>e</CredentialType>
+				<CredentialType>f</CredentialType>
+			</Or>
+		</And>
+	</Dependencies>`)
```

Comment:
Like I said in the meeting, I think it is a little bit confusing that the elements of the `Or` tag form a conjunction. My first impression would be that they form a disjunction then. I would then turn it around and rename the current `<And>` tag to `<Or>` and the current `<Or>` tag to `<And>`.